src/storage/raw/raw_mvcc.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+// #[PerformanceCriticalPath]
+use crate::storage::kv::{Iterator, Result, Snapshot};
+
+use engine_traits::{CfName, DATA_KEY_PREFIX_LEN};
+use engine_traits::{IterOptions, ReadOptions};
+use txn_types::{Key, TimeStamp, Value};
+
+#[derive(Clone)]
+pub struct RawMvccSnapshot<S: Snapshot> {
+    snap: S,
+}
+
+impl<S: Snapshot> RawMvccSnapshot<S> {
+    pub fn from_snapshot(snap: S) -> Self {
+        RawMvccSnapshot { snap }
+    }
+
+    pub fn seek_first_key_value_cf(
+        &self,
+        cf: Option<CfName>,
+        opts: Option<ReadOptions>,
+        key: &Key,
```

Comment:
In `raw_get()` of `storage.rs`, `Key` is constructed by `Key::from_encoded(raw_key)` which is just to reuse the  `Key` struct for raw operations, but the key isn't actually an encoded key. It doesn't matter for rawkv V1, but it is wrong for the mvcc rawkvV2.

Reply:
need to adjust the code in `storage.rs` as well

Reply:
we will use from_raw to encode the apiv2 key as txn kv does. The pr is on the way by@pingyu.

Reply:
Actually we define a struct RawKey to encode apiv2 key and encode timestamp in key. Please ref: https://github.com/tikv/rfcs/pull/86

Reply:
test_raw_mvcc_snapshot is added to cover this file.